frontend/src/views/DeleteViewTemp.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+    <div class="mb-4">
+      <ElButton
+        key="primary"
+        type="text"
+        @click="onClickDelete(roleId)"
+        class="deletebutton"
+      >
+        Delete
+      </ElButton>
+    </div>
+    <ElDialog
+      v-model="dialogVisible"
+      title="Warning"
+      width="500"
+    >
+      <span>Role will be permanently removed. Continue?</span>
+  
+      <template #footer>
+        <div class="dialog-footer">
+          <ElButton @click="onClickCancel()">Cancel</ElButton>
+          <ElButton type="primary" @click="deleteRole()">
+            Confirm
+          </ElButton>
+        </div>
+      </template>
+    </ElDialog>
+  </template>
+  
+  <script setup lang="ts">
+  import { ElButton, ElDialog, ElMessage } from 'element-plus';
+  import { ref } from 'vue'
+  import { deleteRequest } from '@/services/http';
+  import { removeRoleFromState } from '@/stores/roles/state';
+  
+  const dialogVisible = ref(false)
+  const roleId = ref<string>('')
+  
+  const onClickDelete = (id: string) => {
+    dialogVisible.value = true;
+    roleId.value = id; 
+  }
+  
+  const onClickCancel = () => {
+    dialogVisible.value=false;
+  }
+
+async function deleteRole() {
+    try {
```

Comment:
You have to create a function inside actions file. This function will responsible for 

1. calling the backend
2. Removing the role from the state

Here you should just import this action, nothing more

Check in Demo how an action is called, to fetch the roles, and base your work like that